suitcase/tiff_series/__init__.py

```diff
@@ -264,38 +249,30 @@ def event_page(self, doc):
             EventPage document
         '''
         event_model.verify_filled(doc)
-        streamname = self._streamnames[doc['descriptor']]
+        descriptor = self._descriptors[doc['descriptor']]
+        streamname = descriptor.get('name')
         for field in doc['data']:
             for img in doc['data'][field]:
                 # check that the data is 2D, if not ignore it
                 img_asarray = numpy.asarray(img)
                 if img_asarray.ndim == 2:
-                    if self._stack_images:
-                        # create a file for this stream and field if required
-                        if not self._tiff_writers.get(streamname, {}).get(field):
-                            filename = (f'{self._templated_file_prefix}'
-                                        f'{streamname}-{field}.tiff')
-                            file = self._manager.open(
-                                'stream_data', filename, 'xb')
-                            tw = TiffWriter(file, **self._init_kwargs)
-                            self._tiff_writers[streamname][field] = tw
-                        # append the image to the file
-                        tw = self._tiff_writers[streamname][field]
-                        tw.save(img_asarray, *self._kwargs)
+                    # template the file name.
+                    self._templated_file_prefix = self._file_prefix.format(
+                        start=self._start, descriptor=descriptor,
+                        event_page=doc)
```

Comment:
This is going to give you vectors for the other entries I think, not the entry for _this_ event.

Reply:
can you elaborate here @tacaswell, this is literally referencing the `doc` passed into this `self.event_doc` call so I am not sure how it is going to reference other entries. In fact I use `doc` in line 252 above to extract out the current `event_page`'s descriptor `uid`.

Reply:
This method writes a TIFF image per EventPage, when what is wanted as a TIFF image per Event. @tacaswell found one consequence of the current design: `{event_page.data.temperature}` would render as `[T1, T2, ..., TN]` (i.e. a vector of all the entries in the whole EventPage) for an even page comprising N Events.

For TIFF stacks, it makes sense to operate on EventPages because vectorization offers better efficiency. For TIFF series with one TIFF image per "row" (Event) we must operate on Events. We could add a loop through rows in the EventPage, or we could move this logic from `Serializer.event_page` to `Serializer.event`, and put [a block like this](https://github.com/NSLS-II/event-model/blob/a697e8eb805b961bb82a20aaaf585914299ef6be/event_model/__init__.py#L291-L296) in `Serializer.event_page`. I suspect the for-loop approach may be more succinct, but I'm not sure, and I think either approach would be good enough.

Reply:
I hadn't considered this, OK I will update this, probably with the `for` loop options unless their are any major objections. I will hold of on this until Monday to give others time to comment on which options is prefered.

Reply:
OK so for reference, the current solution _does_ save one image per file, contrary to @danielballan's statement above using the loop method, but this does _not_ associate other items from the `event_page` documents with a single value, which is what I suspect @tacaswell was getting at (good catch @tacaswell ). I don't see any solution, that does not introduce major complexity, other than to convert the `event_page` doc to `event` docs. Hence I will implement that solution.

Reply:
resolved